pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -198,6 +201,17 @@ func (r *ReconcileBareMetalHost) Reconcile(request reconcile.Request) (result re
 		}
 	}
 
+	// Check if Status is empty and status annotation is present
+	// Manually restore data.
+	if !r.hostHasStatus(host) {
+		obj := r.getHostFromAnnotation(host)
+		if obj != nil {
+			host.Status = obj.Status
```

Comment:
What causes the host to be stored after this? Shouldn't we be looking at the value of the annotation and the host value, updating host only if there are changes we need, then storing and returning immediately?

Reply:
As, pointed in one of my previous reply, this point is only reached when the status is empty on bmh but the annotation isn't. And storing is caused by the normal reconciliation workflow which checks whether the bmh status is dirty or not.